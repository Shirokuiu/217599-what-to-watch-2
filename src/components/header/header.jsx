import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";

import Avatar from "../avatar/avatar";
import {Breadcrumbs} from "../breadcrumbs/breadcrumbs";

export const Header = (props) => {
  const {parent, movieId, movies} = props;
  const ParentStatus = {
    ADD_REVIEW: `addReview`,
    MY_LIST: `myList`
  };

  return <header className={ParentStatus[parent] === ParentStatus.MY_LIST ? `page-header user-page__head` : `page-header movie-card__head`}>
    <div className="logo">
      <Link to="/" className="logo__link">
        <span className="logo__letter logo__letter--1">W</span>
        <span className="logo__letter logo__letter--2">T</span>
        <span className="logo__letter logo__letter--3">W</span>
      </Link>
    </div>

    {ParentStatus[parent] === ParentStatus.ADD_REVIEW ? <Breadcrumbs movieId={movieId} movies={movies}/> : null}
    {ParentStatus[parent] === ParentStatus.MY_LIST ? <h1 className="page-title user-page__title">My list</h1> : null}

    <Avatar />
  </header>;
};

Header.propTypes = {
  parent: PropTypes.string,
  movieId: PropTypes.number,
  movies: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.array.string,
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
  }))
};
