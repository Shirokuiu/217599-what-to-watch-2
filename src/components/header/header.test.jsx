import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import {BrowserRouter} from "react-router-dom";

import {Header} from "./header";

Enzyme.configure({adapter: new Adapter()});

it(`Is header rendered`, () => {
  const tree = shallow(<BrowserRouter>
    <Header
      isAuthorized={false}
    />
  </BrowserRouter>);

  expect(toJson(tree)).toMatchSnapshot();
});
