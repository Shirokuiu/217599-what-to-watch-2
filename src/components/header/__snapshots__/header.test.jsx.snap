// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is header rendered 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Header
    isAuthorized={false}
  />
</Router>
`;
